starter-analytics-postgres/src/main/java/io/micronaut/starter/analytics/postgres/AnalyticsController.java

```diff
@@ -93,4 +103,20 @@ HttpStatus applicationGenerated(@NonNull @Body Generated generated) {
         featureRepository.saveAll(features);
         return HttpStatus.ACCEPTED;
     }
+
+    /**
+     * Generates an Excel spreadsheet containing all the applications stored in the repository.
+     * @return an Excel spreadsheet.
+     * @throws IOException if the spreadsheet cannot be written to the buffer.
+     */
+    @Get("/excel")
+    @Produces(MediaType.MICROSOFT_EXCEL_OPEN_XML)
+    HttpResponse<StreamedFile> generateExcel() throws IOException {
+        try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+            excelGenerator.generateExcel(applicationRepository.list(Pageable.UNPAGED)).write(out);
```

Comment:
Should we change this code so that we don't load every row into memory but we paginate all the rows instead and build the excel file with 50 rows at a time? 

I want an excel file with every row. But I assume it will be more efficient to do 100 queries to bring 50 rows from the database at a time than a single querie and bring every entry.

Reply:
I just pushed some changes to make page size configurable via `excel.fetch.page.size` -- 50 by default.